src/game/coreObjects/Piece.js

```diff
@@ -57,7 +57,8 @@ class Piece {
   }
 
   // Before the piece is activated to a global location, it shouldn't be used/updated.
-  activatePiece({ center_x, center_y, direction, pidSize, ttl }) {
+  activatePiece({ coreState, center_x, center_y, direction, pidSize, ttl }) {
+    this.mainCell.coreState = coreState;
```

Comment:
If this.mainCell is only ever initialized with a defined coreState, get rid of this line